code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -19,7 +19,22 @@
 	new /obj/effect/temp_visual/telekinesis(get_turf(caller))
 	caller.visible_message("<span class='xenowarning'>A strange buzzing hum starts to emanate from \the [caller]!</span>", \
 	"<span class='xenodanger'>We call forth the larvas to rise from their slumber!</span>")
-	notify_ghosts("\The <b>[caller]</b> is calling for the burrowed larvas to wake up!", enter_link = "join_larva=1", enter_text = "Join as Larva", source = caller, action = NOTIFY_JOIN_AS_LARVA)
+
+	var/burrowed = caller.hive.stored_larva
+	var/total = 0
+	for(var/i in 1 to burrowed)
+		var/mob/M = get_alien_candidate()
+		if(!M)
+			continue

```

Comment:
If it fails to get one, will it be able to succeed the next time?
If not, this could be a `break` instead.